### PR TITLE
Feature/tl2023 212 add assessment entry functionality

### PR DESCRIPTION
**Pull request by sajidmalik**

### CODE REVIEW

File: src/Sfa.Tl.ResultsAndCertification.Models/Contracts/Comparer/AssessmentComparer.cs [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+                return false;
+            
+            return x.SeriesId == y.SeriesId && x.SeriesName == y.SeriesName;
+        }
+
+        public int GetHashCode(PathwayAssessment assessment)
+        {
+            unchecked
+            {
+                var hashCode = assessment.SeriesId.GetHashCode();
```
Comment (ibai-gonzalez) [inline src/Sfa.Tl.ResultsAndCertification.Models/Contracts/Comparer/AssessmentComparer.cs:23]: Wouldn't it be better if you used the ''SeriesName" hashcode too?

File: src/Sfa.Tl.ResultsAndCertification.Web/Controllers/AdminDashboardController.cs [hunk 2]
```diff
@@ -451,8 +451,66 @@ public async Task<IActionResult> AdminReviewChangesIndustryPlacementAsync(AdminR
...
+                return RedirectToRoute(RouteConstants.PageNotFound);
+
+            return View(viewModel);
+        }
+
+        [HttpPost]
+        [Route("admin/submit-add-assessment-entry-core", Name = RouteConstants.SubmitCoreComponentAssessmentEntry)]
+        public async Task<IActionResult> AdminCoreComponentAssessmentEntry(AdminCoreComponentViewModel model)
+        {
+            var adminCoreComponent = await _loader.GetAdminLearnerRecordWithCoreComponents(model.RegistrationPathwayId);
```
Comment (ibai-gonzalez) [inline src/Sfa.Tl.ResultsAndCertification.Web/Controllers/AdminDashboardController.cs:476]: Why do you have call the loader when the model comes as a parameter already?

File: src/Sfa.Tl.ResultsAndCertification.Web/Controllers/AdminDashboardController.cs [hunk 3]
```diff
@@ -451,8 +451,66 @@ public async Task<IActionResult> AdminReviewChangesIndustryPlacementAsync(AdminR
...
+                return RedirectToRoute(RouteConstants.PageNotFound);
+
+            return View(viewModel);
+        }
+
+        [HttpPost]
+        [Route("admin/submit-add-assessment-entry-specialism", Name = RouteConstants.SubmitOccupationalSpecialisAssessmentEntry)]
+        public async Task<IActionResult> AdminOccupationalSpecialismAssessmentEntry(AdminOccupationalSpecialismViewModel model)
+        {
+            var adminOccupationalSpecialism = await _loader.GetAdminLearnerRecordWithOccupationalSpecialism(model.RegistrationPathwayId, model.SpecialismAssessmentId);
```
Comment (ibai-gonzalez) [inline src/Sfa.Tl.ResultsAndCertification.Web/Controllers/AdminDashboardController.cs:503]: Same as the other POST method

**Review by Skumarhpa — APPROVED** [review]

File: src/Sfa.Tl.ResultsAndCertification.Web/ViewModel/AdminDashboard/AdminComponentExamViewModel.cs [hunk 4]
```diff
@@ -0,0 +1,41 @@
+﻿using Sfa.Tl.ResultsAndCertification.Common.Enum;
```
Comment (ibai-gonzalez) [inline src/Sfa.Tl.ResultsAndCertification.Web/ViewModel/AdminDashboard/AdminComponentExamViewModel.cs:1]: This file can be removed. It is not used anywhere. It was taken from my old branch sorry!

File: src/Sfa.Tl.ResultsAndCertification.Web/Controllers/AdminDashboardController.cs (hunk 2 above)
Comment (sajidmalik) [reply src/Sfa.Tl.ResultsAndCertification.Web/Controllers/AdminDashboardController.cs:476]: There is data on the model which is missing on the post - which can be added to the form as hidden items or in the cache. However that can impact the page performance. This  seem to be a better way to implement.

File: src/Sfa.Tl.ResultsAndCertification.Web/Controllers/AdminDashboardController.cs (hunk 3 above)
Comment (sajidmalik) [reply src/Sfa.Tl.ResultsAndCertification.Web/Controllers/AdminDashboardController.cs:503]: There is data on the model which is missing on the post - which can be added to the form as hidden items or in the cache. However that can impact the page performance. This  seem to be a better way to implement.

**Pull request merged**